htdocs/product/class/html.formproduct.class.php

```diff
@@ -235,15 +235,37 @@ private function get_parent_path($tab, $final_label = '')
      *
      *  @throws Exception
 	 */
-	public function selectWarehouses($selected = '', $htmlname = 'idwarehouse', $filterstatus = '', $empty = 0, $disabled = 0, $fk_product = 0, $empty_label = '', $showstock = 0, $forcecombo = 0, $events = array(), $morecss = 'minwidth200', $exclude = '', $showfullpath = 1, $stockMin = false, $orderBy = 'e.ref')
+	public function selectWarehouses($selected = '', $htmlname = 'idwarehouse', $filterstatus = '', $empty = 0, $disabled = 0, $fk_product = 0, $empty_label = '', $showstock = 0, $forcecombo = 0, $events = array(), $morecss = 'minwidth200', $exclude = array(), $showfullpath = 1, $stockMin = false, $orderBy = 'e.ref')
 	{
-		global $conf,$langs,$user;
+		global $conf,$langs,$user, $hookmanager;
 
 		dol_syslog(get_class($this)."::selectWarehouses $selected, $htmlname, $filterstatus, $empty, $disabled, $fk_product, $empty_label, $showstock, $forcecombo, $morecss", LOG_DEBUG);
 
 		$out='';
 		if (empty($conf->global->ENTREPOT_EXTRA_STATUS)) $filterstatus = '';
         if (!empty($fk_product))  $this->cache_warehouses = array();
+
+        $parameters = array(
+            'selected' => & $selected,
```

Comment:
Currently the $parameter array for executeHooks is designed to be a readonly array of information of context.
do you really need 
'selected' => & $selected
instead of 
'selected' => $selected

Reply:
I need to modify the "exclude" field.

I think to several solutions : 
- & only for $exclude
- place hook after "$this->loadWarehouses" to do it in hook and replace it

Which solution is the best according to you ?

Reply:
I see. 
If you need to change the $exclude, it means you probably want to change the result of the $out that is forged later ? Am i right ?
If yes, a better solution is a hook after.

You suggest a hook after the loadWarehouses(). But what do you think about a hook after the $out is forged, completely before the $out. So the module will be able not only to replace the result of select, but also how it is rendered for output. I will accept both if the hook is after the loadWarehouses (archi is ok in both case) but i would prefer to have hook just before the return for more possibilities.

Reply:
It's ok for me. I moved hook just before the "return $out" for more possibilities.